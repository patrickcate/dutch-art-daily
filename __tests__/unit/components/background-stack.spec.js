import { shallowMount } from '@vue/test-utils'
import mockData from '@fixtures/mock-data.js'
import BackgroundStack from '@components/background-stack.vue'

describe('BackgroundStack Component', () => {
  it('is a Vue instance', () => {
    const wrapper = shallowMount(BackgroundStack, {
      mocks: {
        $store: {
          state: {
            currentPage: '01-01',
            loadedSlides: [],
          },
        },
      },
    })

    expect(wrapper.isVueInstance()).toBe(true)
  })

  it('renders correctly', () => {
    const wrapper = shallowMount(BackgroundStack, {
      mocks: {
        $store: {
          state: {
            currentPage: '01-01',
            loadedSlides: ['01-01'],
          },
          getters: {
            getArtworkById() {
              return mockData.artwork
            },
          },
        },
      },
    })

    expect(wrapper).toMatchSnapshot()
  })

  it('does not render correctly', () => {
    const wrapper = shallowMount(BackgroundStack, {
      mocks: {
        $store: {
          state: {
            currentPage: '01-01',
            loadedSlides: [],
          },
        },
      },
    })

    expect(wrapper).toMatchSnapshot()
  })

  it('has no active background', () => {
    const wrapper = shallowMount(BackgroundStack, {
      mocks: {
        $store: {
          state: {
            currentPage: '01-01',
            loadedSlides: ['01-01'],
          },
          getters: {
            getArtworkById() {
              return null
            },
          },
        },
      },
    })

    expect(wrapper).toMatchSnapshot()
  })
})
