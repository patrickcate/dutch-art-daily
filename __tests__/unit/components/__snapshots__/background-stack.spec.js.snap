// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackgroundStack Component does not render correctly 1`] = ``;

exports[`BackgroundStack Component has no active background 1`] = `
<div class="background-stack">
  <background-stack-item-stub id="01-01"></background-stack-item-stub>
</div>
`;

exports[`BackgroundStack Component renders correctly 1`] = `
<div class="background-stack" style="background-color: rgb(101, 72, 49); background-image: url(/photos/01-01/01-01--lqi-d0wAmzwM620.jpg);">
  <background-stack-item-stub id="01-01"></background-stack-item-stub>
</div>
`;
